core/drivers/clk/clk-stm32mp13.c

```diff
@@ -0,0 +1,1727 @@
+		/* Same parameters, no need to config */
+		*result = 1;
+	else
+		*result = 0;
+
+	return 0;
+}
+
+static int stm32_clk_hsidiv_configure(struct clk_stm32_priv *priv)
+{
+	struct stm32_clk_platdata *pdata = priv->pdata;
+	struct stm32_osci_dt_cfg *osci = &pdata->osci[OSC_HSI];
+
+	return clk_stm32_set_rate_divider(DIV_HSI, osci->freq, MAX_HSI_HZ);
+}
+
+static void clk_stm32_pll_config_vco(struct clk_stm32_priv *priv,
+				     const struct stm32_clk_pll *pll,
+				     struct stm32_pll_vco *vco)
+{
+	uintptr_t pll_base = priv->base + pll->reg_pllxcr;
+	uint32_t value = 0;
+
+	if (clk_stm32_pll_compute_cfgr1(pll, vco, &value) != 0) {
+		EMSG("Invalid Vref clock");
+		panic();
+	}
+
+	/* Write N / M / IFREGE fields */
+	io_write32(pll_base + RCC_OFFSET_PLLXCFGR1, value);
+
+	/* Fractional configuration */
+	io_write32(pll_base + RCC_OFFSET_PLLXFRACR, 0);
+
+	/* Frac must be enabled only once its configuration is loaded */
+	io_write32(pll_base + RCC_OFFSET_PLLXFRACR,
+		   vco->frac << RCC_PLLNFRACR_FRACV_SHIFT);
+
+	io_setbits32(pll_base + RCC_OFFSET_PLLXFRACR, RCC_PLLNFRACR_FRACLE);
+}
+
+static void clk_stm32_pll_config_csg(struct clk_stm32_priv *priv,
+				     const struct stm32_clk_pll *pll,
+				     struct stm32_pll_vco *vco)
+{
+	uintptr_t pll_base = priv->base + pll->reg_pllxcr;
+	uint32_t mod_per = 0;
+	uint32_t inc_step = 0;
+	uint32_t sscg_mode = 0;
+	uint32_t value = 0;
+
+	if (!vco->csg_enabled)
+		return;
+
+	mod_per = vco->csg[PLL_CSG_MOD_PER];
+	inc_step = vco->csg[PLL_CSG_INC_STEP];
+	sscg_mode = vco->csg[PLL_CSG_SSCG_MODE];
+
+	value |= (mod_per << RCC_PLLNCSGR_MOD_PER_SHIFT) &
+		 RCC_PLLNCSGR_MOD_PER_MASK;
+	value |= (inc_step << RCC_PLLNCSGR_INC_STEP_SHIFT) &
+		 RCC_PLLNCSGR_INC_STEP_MASK;
+	value |= (sscg_mode << RCC_PLLNCSGR_SSCG_MODE_SHIFT) &
+		 RCC_PLLNCSGR_SSCG_MODE_MASK;
+
+	io_write32(pll_base + RCC_OFFSET_PLLXCSGR, value);
+	io_setbits32(pll_base + RCC_OFFSET_PLLXCR, RCC_PLLNCR_SSCG_CTRL);
+}
+
+static void clk_stm32_pll_config_out(struct clk_stm32_priv *priv,
+				     const struct stm32_clk_pll *pll,
+				     struct stm32_pll_output *out)
+{
+	uintptr_t pll_base = priv->base + pll->reg_pllxcr;
+	uint32_t value = 0;
+
+	value = clk_stm32_pll_compute_cfgr2(out);
+
+	io_write32(pll_base + RCC_OFFSET_PLLXCFGR2, value);
+}
+
+static inline struct stm32_pll_dt_cfg *clk_stm32_pll_get_pdata(int pll_idx)
+{
+	struct clk_stm32_priv *priv = clk_stm32_get_priv();
+	struct stm32_clk_platdata *pdata = priv->pdata;
+
+	return &pdata->pll[pll_idx];
+}
+
+static int clk_stm32_pll_init_switch_to_hsi_clk_system(int mux_sys)
+{
+	int sel = 0;
+
+	if (mux_sys == -1)
+		return -1;
+
+	/* Make a backup to the current parent */
+	sel = clk_stm32_get_parent_mux(mux_sys);
+
+	/* Switch to HSI */
+	clk_stm32_set_parent_mux(mux_sys, 0);
+
+	return sel;
+}
+
+static void clk_stm32_pll_init_restore_clk_system(int mux_sys, int sel)
+{
+	if (mux_sys != -1) {
+		/* Restore the parent */
+		clk_stm32_set_parent_mux(mux_sys, sel);
```

Comment:
check return value

Reply:
clk_stm32_pll_init_restore_clk_system() is used only one time, i will move this code and made the check